lib/core/client.js

```diff
@@ -443,7 +443,13 @@ class NightwatchClient extends EventEmitter {
   }
 
   setLaunchUrl() {
-    const value = this.settings.baseUrl || this.settings.launchUrl || this.settings.launch_url || null;
+    let value = this.settings.baseUrl || this.settings.launchUrl || this.settings.launch_url || null;
+
+    // For e2e and component testing on android emulator
+    if (this.settings.desiredCapabilities.avd) {         
+      value = value.replace('localhost', '10.0.2.2');
```

Comment:
You should also consider the instance when someone has used `127.0.0.1` instead of `localhost`.

```suggestion
      value = value.replace(/localhost|127.0.0.1/g, '10.0.2.2');
```